src/components/MultiHopTrade/hooks/useAccountIds.tsx

```diff
@@ -12,21 +16,28 @@ export const useAccountIds = (): {
 } => {
   const dispatch = useAppDispatch()
 
+  const walletId = useAppSelector(selectWalletId)
   // Default sellAssetAccountId selection
   const sellAssetAccountId = useAppSelector(selectFirstHopSellAccountId)
   const buyAssetAccountId = useAppSelector(selectLastHopBuyAccountId)
 
+  // Resets the sell and buy asset AccountIDs on wallet change to that we don't get stale trade input account selections while we're loading the new wallet
+  useEffect(() => {
```

Comment:
This `useEffect` does get called on every `/trade` route mount, but that's fine as the accountIds are still correctly set later.

_However_, this does mean that we no longer remember the selected accounts when clicking "Preview Trade", and then navigating back to the quote screen. This is a regression from prod that we may or may not want to accept.

Reply:
Nice catch! Thought all that was needed to stop the bleed here was to move this to `<AppContext />`... however still no luck, as, even though this effect doesn't run anymore, the `onChange` events of `<AccountDropdown />` in `<TradeInputBody />` fire and reset the AccountIds.
Another bug I've noted while trying to fix this is clicking "Trade" from an asset page doesn't set the buy asset proper anymore. Stay tuned for resolution in this PR @0xApotheosis 🙏🏽

Reply:
Aaand fixed in https://github.com/shapeshift/web/pull/7916/commits/a0d1fbf595ae1d1e1d71ea136b659ef97ec1960d / https://github.com/shapeshift/web/pull/7916/commits/898715d6b2b9ad331d2eb25fbea88f3346dccd24 @0xApotheosis!

Localhost Jams: 
- AccountIds are still remembered: https://jam.dev/c/2139fbe8-e5c0-43f4-b2db-59f0282a6f38
- AccountIds are reset on wallet change: https://jam.dev/c/76eb4e93-6bbf-467a-96c2-e1cbf2283517

Note, the weird issue re: asset page routing is also present on current develop locally for some weird reason, but is happy upstream. This confirms that this isn't introduced by this diff, deployed to gome as a paranoia and retested there:

- Asset Page routing is happy: https://jam.dev/c/31331f51-6531-4c87-8526-7d2b536c2218
- AccountIds are still remembered: https://jam.dev/c/a499bc38-c691-4192-8e31-d42030889048
- AccountIds are reset on wallet change: https://jam.dev/c/e7147f5a-bb21-4f1b-bb92-53cf949e527d